### PR TITLE
Storeage/STG95 Files Binary ACE

### PR DESCRIPTION
**Pull request by microzchang**
# Pull Request Checklist

Please leverage this checklist as a reminder to address commonly occurring feedback when submitting a pull request to make sure your PR can be reviewed quickly:

See the detailed list in the [contributing guide](https://github.com/Azure/azure-sdk-for-cpp/blob/main/CONTRIBUTING.md#pull-requests).

- [ ] [C++ Guidelines](https://azure.github.io/azure-sdk/cpp_introduction.html)
- [ ] Doxygen docs
- [ ] Unit tests
- [ ] No unwanted commits/changes
- [ ] Descriptive title/description
  - [ ] PR is single purpose
  - [ ] Related issue listed
- [ ] Comments in source
- [ ] No typos
- [ ] Update changelog
- [ ] Not work-in-progress
- [ ] External references or docs updated
- [ ] Self review of PR done
- [ ] Any breaking changes?

### CODE REVIEW

**Comment by microzchang** [comment]
In dotnet, they overload a CreatePermission and directly use `ShareFilePermission` as parameter. Unsure if we should do this in C++. 
Directly add `FilePermssionFormat` can avoid overload, but this may be unconvenient for customer.

@Jinming-Hu Do you have any idea about this?

File: sdk/storage/azure-storage-files-shares/inc/azure/storage/files/shares/share_client.hpp [hunk 1]
```diff
@@ -242,6 +242,23 @@ namespace Azure { namespace Storage { namespace Files { namespace Shares {
...
+     * the format in which the permission is returned.If filePermissionKeyFormat is unspecified or
+     * explicityly set to SSDL format, the permission will be
+     * returned in SSDL format.
+     * @param options Optional parameters to get share's permission.
+     * @param context Context for cancelling long running operations.
+     * @return Azure::Response<std::string> containing the permission string with specified key.
+     */
+    Azure::Response<Models::ShareFilePermission> GetPermission(
+        const std::string& permissionKey,
+        const Nullable<Models::FilePermissionFormat> filePermissionFormat,
```
Comment (Jinming-Hu) [inline sdk/storage/azure-storage-files-shares/inc/azure/storage/files/shares/share_client.hpp:258]: why is this optional parameter special that it cannot go into option bag?

File: sdk/storage/azure-storage-files-shares/inc/azure/storage/files/shares/share_client.hpp [hunk 2]
```diff
@@ -242,6 +242,23 @@ namespace Azure { namespace Storage { namespace Files { namespace Shares {
...
+     * @param permissionKey The permission key of a permission.
+     * @param filePermissionFormat Optional. Available for version 2024-11-04 and later. Specifies
+     * the format in which the permission is returned.If filePermissionKeyFormat is unspecified or
+     * explicityly set to SSDL format, the permission will be
+     * returned in SSDL format.
+     * @param options Optional parameters to get share's permission.
+     * @param context Context for cancelling long running operations.
+     * @return Azure::Response<std::string> containing the permission string with specified key.
+     */
+    Azure::Response<Models::ShareFilePermission> GetPermission(
```
Comment (Jinming-Hu) [inline sdk/storage/azure-storage-files-shares/inc/azure/storage/files/shares/share_client.hpp:256]: can we just use Response<string> as return type?

Comment (microzchang) [reply sdk/storage/azure-storage-files-shares/inc/azure/storage/files/shares/share_client.hpp:256]: You mean wrapper the return value with a response type such as `GetPermissionResponse` or
we just use Response<std::string> which like previous one and don't overload this method? In this impl, user have to get the FilePermissionFormat from RawResponse.

**Review by Jinming-Hu — APPROVED** [review]

**Pull request merged**